### PR TITLE
Move plutus-example from cardano-node to plutus-apps

### PR DESCRIPTION
**Pull request by Jimbo4350**

### CODE REVIEW

File: doc/plutus/howtos/plutus-spending-script-example.rst [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+Run the following commands:
+
+.. code:: bash
+
+   cd plutus-example
+
+   cabal run exe:plutus-example
+
+This will output ``always-succeeds-txin.plutus`` in the
+``generated-plutus-scripts`` dir.
```
Comment (newhoggy) [inline doc/plutus/howtos/plutus-spending-script-example.rst:71]: Should `generated-plutus-scripts` be `.gitignored`?

File: doc/plutus/howtos/plutus-spending-script-example.rst [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+There is a convenient script that will set up an Alonzo cluster
+immediately on your local machine.
+
+Run the following command:
+
+.. code:: bash
+
+   cabal install cardano-cli
+   cabal install cardano-node
+   ./scripts/byron-to-alonzo/mkfiles.sh alonzo
```
Comment (newhoggy) [inline doc/plutus/howtos/plutus-spending-script-example.rst:85]: The `mkfiles.sh` file appears to not exist.

File: doc/plutus/howtos/plutus-spending-script-example.rst [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+   > cardano-cli transaction hash-script-data --script-data-value 42
+   > 9e1199a988ba72ffd6e9c269cadb3b53b5f360ff99f112d9b2ee30c4d74ad88b
+
+In this example, the script we are using always succeeds so we can use
+any datum hash. We calculate the script address as follows:
+
+.. code:: bash
+
+   > cardano-cli address build --payment-script-file plutus-example/generated-plutus-scripts/always-succeeds-txin.plutus  --testnet-magic 42
```
Comment (newhoggy) [inline doc/plutus/howtos/plutus-spending-script-example.rst:107]: `plutus-example/generated-plutus-scripts/always-succeeds-txin.plutus` does not exist.

File: doc/plutus/howtos/plutus-spending-script-example.rst [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+.. code:: bash
+
+   cardano-cli transaction build-raw \
+     --alonzo-era \
+     --fee 0 \
+     --tx-in $txin \
+     --tx-out "addr_test1wzeqkp6ne3xm6gz39l874va4ujgl4kr0e46pf3ey8xsu3jsgkpcj2+$lovelace" \
+     --tx-out-datum-hash 9e1199a988ba72ffd6e9c269cadb3b53b5f360ff99f112d9b2ee30c4d74ad88b \
+     --out-file create-datum-output.body
```
Comment (newhoggy) [inline doc/plutus/howtos/plutus-spending-script-example.rst:121]: `$txin` does reference anything?

Comment (newhoggy) [reply doc/plutus/howtos/plutus-spending-script-example.rst:121]: Same for some other variables mentioned in this file.

File: plutus-example/plutus-example.cabal [hunk 5]
```diff
@@ -0,0 +1,155 @@
+cabal-version: 3.0
+
+name:                   plutus-example
+version:                1.33.0
```
Comment (newhoggy) [inline plutus-example/plutus-example.cabal:4]: What is our versioning policy now that this has moved out of `cardano-node`?  Should it be updated to follow `cardano-node`'s latest version or is this independently versioned now?

File: plutus-example/plutus-example.cabal [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+                      , plutus-ledger
+                      , plutus-ledger-api
+                      , tasty
+                      , tasty-hedgehog
+                      , text
+                      , unordered-containers
+
+
+  other-modules:
+                        -- TODO: Re-enable when cardano-node is bumped to the latest master
```
Comment (newhoggy) [inline plutus-example/plutus-example.cabal:144]: Is this TODO relevant?

File: plutus-example/src/PlutusExample/ScriptContextChecker.hs [hunk 7]
```diff
@@ -0,0 +1,433 @@
...
+  Plutus.txInfoValidRange txInfo P.== txValidRange P.&&
+  -- Datums and datum hashes are equivalent
+  Plutus.txInfoData txInfo P.== datumsAndHashes P.&&
+  -- Required tx signers are equivalent
+  Plutus.txInfoSignatories txInfo P.== signatories P.&&
+  -- Payment tx out is equivalent
+  AMap.member paymentOutputFromRedeemer scriptContextOutputsMap P.&&
+  -- Txins are equivalent
+  (AMap.member txinA scriptContextTxinsMap P.&& AMap.member txinB scriptContextTxinsMap) P.&&
+  -- Cert if equivalent
```
Comment (newhoggy) [inline plutus-example/src/PlutusExample/ScriptContextChecker.hs:110]: Not clear to me what "cert if equivalent" means.  Does it mean "certify if equivalent"?  Or "certificate if equivalent?

File: plutus-example/test/Test/PlutusExample/Gen.hs [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+import Cardano.Api
+import Cardano.Api.Shelley
+import Prelude
+
+import Data.Map.Strict qualified as Map
+
+import Cardano.Ledger.Alonzo.Tx qualified as Alonzo
+import Cardano.Ledger.Alonzo.TxInfo qualified as Alonzo
+-- import           Cardano.Ledger.BaseTypes (ProtoVer)
```
Comment (newhoggy) [inline plutus-example/test/Test/PlutusExample/Gen.hs:13]: Should this comment be deleted?

File: plutus-example/plutus-example.cabal [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+  main-is:              create-script-context.hs
+  ghc-options:          -threaded -rtsopts "-with-rtsopts=-T"
+
+  build-depends:        bytestring
+                      , cardano-api
+                      , optparse-applicative
+                      , plutus-example
+                      , transformers
+
+test-suite plutus-example-test
```
Comment (newhoggy) [inline plutus-example/plutus-example.cabal:114]: Is this currently being run in CI?

File: doc/plutus/howtos/plutus-spending-script-example.rst (hunk 1 above)
Comment (Jimbo4350) [reply doc/plutus/howtos/plutus-spending-script-example.rst:71]: Yep :+1:

File: doc/plutus/howtos/plutus-spending-script-example.rst (hunk 2 above)
Comment (Jimbo4350) [reply doc/plutus/howtos/plutus-spending-script-example.rst:85]: Hmmm....I'm going to point to the script in `cardano-node`. Potentially maintaining two scripts I think is a bad idea.

File: doc/plutus/howtos/plutus-spending-script-example.rst (hunk 4 above)
Comment (Jimbo4350) [reply doc/plutus/howtos/plutus-spending-script-example.rst:121]: I linked to this script to see how variables are assigned: https://github.com/input-output-hk/cardano-node/blob/master/scripts/plutus/example-txin-locking-plutus-script.sh

File: plutus-example/plutus-example.cabal (hunk 6 above)
Comment (Jimbo4350) [reply plutus-example/plutus-example.cabal:144]: Yeah, `plutus-apps` is pinned to a cardano-node commit in December

File: plutus-example/test/Test/PlutusExample/Gen.hs (hunk 8 above)
Comment (Jimbo4350) [reply plutus-example/test/Test/PlutusExample/Gen.hs:13]: Yep, copy paste error

File: plutus-example/plutus-example.cabal (hunk 5 above)
Comment (Jimbo4350) [reply plutus-example/plutus-example.cabal:4]: @asutherlandus

File: plutus-example/plutus-example.cabal (hunk 9 above)
Comment (Jimbo4350) [reply plutus-example/plutus-example.cabal:114]: @asutherlandus  How do I add this to run in CI?

**Review by newhoggy — APPROVED** [review]

File: plutus-example/README.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+## FAQ
+
+### Where is the off chain code?
+
+The off chain code is used for transaction construction. In this case we construct the transaction with `cardano-cli` and therefore we don't need to write any off chain code.
+
+### Where can I learn about Plutus scripts in more detail?
+
+Our education director, Lars Brünjes, has an excellent series of [tutorials](https://youtu.be/IEn6jUo-0vU) on youtube. We will not attempt to provide an indepth explanation of [Plutus](https://docs.cardano.org/projects/plutus/en/latest/) in this repository.
```
Comment (james-iohk) [inline plutus-example/README.md:20]: We should update Youtube link for 2022
Iteration 3 is made up of multiple playlists, this is the best link I could find: https://www.youtube.com/channel/UCX9j__vYOJu00iqBrCzecVw/playlists?view=50&sort=dd&shelf_id=2

Comment (james-iohk) [reply plutus-example/README.md:20]: The statement "We will not attempt to provide an indepth explanation of Plutus in this repository" may no longer be true. (the docs.cardano.org link is also dead).
We can refer to either `doc/plutus/` or https://plutus-apps.readthedocs.io/en/latest/

File: plutus-example/plutus-example.cabal (hunk 9 above)
Comment (koslambrou) [reply plutus-example/plutus-example.cabal:114]: I did run in CI: https://hydra.iohk.io/build/12955862

File: plutus-example/README.md (hunk 10 above)
Comment (Jimbo4350) [reply plutus-example/README.md]: Thanks @james-iohk, I made the changes.

**Pull request merged**